kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/StatsUtils.java

```diff
@@ -44,14 +53,167 @@ public static Optional<DataFileStatistics> deserializeFromJson(String json) {
       throw new KernelException(String.format("Failed to parse JSON string: %s", json), e);
     }
 
-    JsonNode numRecordsNode = root.get("numRecords");
+    // Parse numRecords
+    JsonNode numRecordsNode = root.get(StatsSchemaHelper.NUM_RECORDS);
     if (numRecordsNode == null || !numRecordsNode.isNumber()) {
       return Optional.empty();
     }
-
     long numRecords = numRecordsNode.asLong();
-    return Optional.of(
-        new DataFileStatistics(
-            numRecords, Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap()));
+
+    // Parse minValues
+    Map<Column, Literal> minValues = new HashMap<>();
+    JsonNode minNode = root.get(StatsSchemaHelper.MIN);
+    if (minNode != null && minNode.isObject()) {
+      parseNestedJsonValues(minNode, minValues, new String[0], true);
+    }
+
+    // Parse maxValues
+    Map<Column, Literal> maxValues = new HashMap<>();
+    JsonNode maxNode = root.get(StatsSchemaHelper.MAX);
+    if (maxNode != null && maxNode.isObject()) {
+      parseNestedJsonValues(maxNode, maxValues, new String[0], true);
+    }
+
+    // Parse nullCount
+    Map<Column, Long> nullCount = new HashMap<>();
+    JsonNode nullCountNode = root.get(StatsSchemaHelper.NULL_COUNT);
+    if (nullCountNode != null && nullCountNode.isObject()) {
+      parseNestedJsonValues(nullCountNode, nullCount, new String[0], false);
+    }
+
+    return Optional.of(new DataFileStatistics(numRecords, minValues, maxValues, nullCount));
+  }
+
+  /**
+   * Helper method to recursively parse nested JSON values back into Column->Value maps. This is the
+   * inverse of the writeJsonValues method in DataFileStatistics.serializeAsJson.
+   *
```

Comment:
nit: could you please add an example Json tree of minValues/maxValues? just one top level value and one nested value.